### ISSUE TITLE
Help understanding set up

### ISSUE THREAD

**Issue by cranesandcaff** [iam4x/isomorphic-flux-boilerplate#45]
I'm sorry, this isn't an issue exactly. 

I've been trying for a few hours to wrap my head around how this boilerplate is set up and I am just completely confused. 

I wanted to add auth and an API. It seemed like I'd need to set up some new routes so I tried using Koa router and the mount middleware the same way the assets are being loaded. But I kept 404ing on the URL.

So I played around with it, commented things out, added dev logs, and it seemed like none of my logs were reaching the console. I copied/pasted one of the existing dev calls a few times and only one went through. Changed text etc. Stopped and restarted the dev server.

Putting a console.log outside of a function, at the root of the file would send it to the console but if I put it inside of any existing or new function it would just disappear.

I commented out nearly everything in the koa.js file and somehow the default app was still being served. I assume this is either caching, or proxying from WebPack. 

How would I go about adding new routes API routes, not component routes?

It doesn't seem like the webpack config is really doing anything crazy so I just feel silly asking this.

If I figure it out on my own I'll close the issue and either comment with a little how-to or put a pull-request with a new doc section if you'd like.

**Comment by cranesandcaff** [comment]
To explain, hopefully a little more of my confusion.

I decided to just create a new express app and server it from a different port. It seems the existing guides favor connecting to an API server so that made sense. If I attempt to include the express app and start it within koa.js it simply doesn't happen. It doesn't exist. Calling `expressApp.listen(9000)` does nothing inside of koa.js, or as called on the file being imported but if I put the listen back in the expressApp.js I wrote and then do a more standard `$ node run server/expressApp.js` alongside the webpack dev server it'll run fine. 

In addition, calling an early `app.use()` that yields the body and ends the response seems to be ignored. Adding any middleware to the Koa app seems to be ignored.

I'm just trying to understand because this has been my favorite looking flux boilerplate, but I just can't seem to understand how it works.

**Comment by adamjv90** [comment]
Hey, just curious how you are starting this app? It is very odd that the app still runs after deleting everything inside server/koa.js .

To answer your question about adding api routes tryin first installing koa-route

```
npm install --save koa-route   
```

then require this module inside server/koa.js

```
import route from 'koa-route';
```

and add routes like so:

```
function *list() {
  var res = [{id: 0, name: 'adam'}, {id:1, name: 'test'}];
  this.body = res;
}

app.use(route.get('/api', list));
```

**Comment by cranesandcaff** [comment]
Yeah it's driving me bonkers. I'm running `npm run dev`. If I commented out the HBS config it wouldn't render, but somehow commenting out the `app.listen()` would still serve it. I realize that sounds like absolute crazy talk. 

I'll give that a try and let you know. Logically it should work.

**Comment by cranesandcaff** [comment]
Yep. I cannot imagine what I was doing wrong. Thanks, sorry for the irrelevant  ticket.

**Comment by adamjv90** [comment]
Have you given the debug package a try for logging development messages? To give you a quick overview of how that package works you simply require it (server/koa.js should already be requiring it)
```
import debug from 'debug';
```

then send it logs:
```
debug('app')('enter log message here');
```
take note of 'app' if you want any messages to log in the console you need to set an environment variable before launching the server script
```
DEBUG=koa,dev,app,app:* npm run dev
```
once you have the debug environment variable set you should start seeing these log both in the console and in the browser console for scripts that would run on the browser. for more info on this package checkout https://www.npmjs.com/package/debug

happy coding!!

**Comment by cranesandcaff** [comment]
Honestly, I'm going to try and save myself a bit and chalk it up to me doing something stupid. I had used the debug method and wasn't able to get it to log with that. New file level logs worked but none with the debug package or inside of the functions.

If I figure out what I did to cause it's weird behavior I'll comment just to share because I'm sure it was something small. 

Thanks for the help! 

Everything is running smoothly.
